common/workunit/workunit.cpp

```diff
@@ -2988,10 +2971,20 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                 case WUStateDebugRunning:
                 case WUStateBlocked:
                 case WUStateAborting:
-                    SessionId agent = conn->queryRoot()->getPropInt64("@agentSession", -1);
-                    if (checkAbnormalTermination(wuid, ret, agent))
+                    if (queryDaliServerVersion().compare("2.1")>=0)
                     {
-                        return ret;
+                        SessionId agent = conn->queryRoot()->getPropInt64("@agentSession", -1);
+                        if((agent>0) && querySessionManager().sessionStopped(agent, 0))
+                        {
+                            conn->reload();
+                            // MCK - does checking state after sessionStopped reveal meaningful info ?
+                            WUState finalState = (WUState) getEnum(conn->queryRoot(), "@state", states);
+                            if ( (WUStateCompleted != finalState) &&
+                                 (WUStateFailed    != finalState) &&
+                                 (WUStateAborted   != finalState) )
```

Comment:
I was suggesting a "continue;" statement after your reload() (line 2979), so that it looped around immediately, no delay.
So, first it would check to see if agentSessionStopped had been set, otherwise it would set and loop (with continue;)

i.e.:
```
    if (agentSessionStopped) // marked true on previous pass
    {
        reportAbnormalTermination(wuid, ret, state, agent);
        return ret;
    }
    agentSessionStopped = true;
    conn->reload();
    continue; // loop back and verify state after session has stopped.
```